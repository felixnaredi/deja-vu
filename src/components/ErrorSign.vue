<!-- eslint-disable -->
<template>
  <div
    class="
      p-8
      bg-gradient-to-t
      from-amber-200
      to-amber-100
      border-solid border-2 border-sky-500
      transition
      ease-in
      opacity-90
      duration-500
      hover:opacity-100
    "
  >
    <h1 class="text-red-500 font-bold text-lg mb-3">Error</h1>
    <div class="mb-16">
      <slot>{{ message }}</slot>
    </div>
    <span
      class="
        border-2
        p-4
        py-3
        border-amber-400
        hover:border-amber-500 hover:bg-amber-300/[.25]
        cursor-default
        select-none
      "
      @click="goHome"
      >D'accueil</span
    >
  </div>
</template>

<script>
import path from "@/service/path";
export default {
  props: {
    message: String,
  },
  methods: {
    goHome: () => {
      window.location.href = path(process.env.BASE_URL);
    },
  },
};
</script>
