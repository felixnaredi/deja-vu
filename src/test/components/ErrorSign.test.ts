import ErrorSign from "@/components/ErrorSign.vue";
import { mount } from "@vue/test-utils";

describe("unit tests", () => {
  test("`ErrorSign` contains the expected slot", () => {
    const wrapper = mount(ErrorSign, {
      slots: {
        default: "<p>1330050845120928966</p>",
      },
    });
    expect(wrapper.find("p").text()).toBe("1330050845120928966");
  });

  test("`goHome` is triggered when button is clicked", async () => {
    const wrapper = mount(ErrorSign);
    wrapper.vm.goHome = vi.fn();
    await wrapper.find("button").trigger("click");
    expect(wrapper.vm.goHome).toBeCalled();
  });

  test("`goHome` sets `window.location.href` to `BASE_URL`", async () => {
    global.window = Object.create(window);
    Object.defineProperty(window, "location", { value: { href: "INITIAL" } });

    const wrapper = mount(ErrorSign);
    await wrapper.find("button").trigger("click");

    // TODO:
    //   Configure so that `import.meta.env.BASE_URL` does not show as an error in vscode.
    // @ts-ignore
    expect(window.location.href).toBe(import.meta.env.BASE_URL);
  });
});

describe("snapshots", () => {
  test("snapshot of empty slot", () => {
    const wrapper = mount(ErrorSign);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("snapshot of realistic usage of slot", () => {
    const wrapper = mount(ErrorSign, {
      slots: {
        default: "<p>This is a test. Do not panic</p>",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
