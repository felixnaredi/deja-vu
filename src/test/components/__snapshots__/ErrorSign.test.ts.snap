// Vitest Snapshot v1

exports[`snapshots > snapshot of empty slot 1`] = `
"<div class=\\"p-8 bg-gradient-to-t from-amber-200 to-amber-100 border-solid border-2 border-sky-500 transition ease-in opacity-90 duration-500 hover:opacity-100\\">
  <h1 class=\\"text-red-500 font-bold text-lg mb-3\\">Error</h1>
  <div class=\\"mb-16\\"></div><button class=\\"border-2 p-4 py-3 border-amber-400 hover:border-amber-500 hover:bg-amber-300/[.25] cursor-default select-none\\"> D'accueil </button>
</div>"
`;

exports[`snapshots > snapshot of realistic usage of slot 1`] = `
"<div class=\\"p-8 bg-gradient-to-t from-amber-200 to-amber-100 border-solid border-2 border-sky-500 transition ease-in opacity-90 duration-500 hover:opacity-100\\">
  <h1 class=\\"text-red-500 font-bold text-lg mb-3\\">Error</h1>
  <div class=\\"mb-16\\">
    <p>This is a test. Do not panic</p>
  </div><button class=\\"border-2 p-4 py-3 border-amber-400 hover:border-amber-500 hover:bg-amber-300/[.25] cursor-default select-none\\"> D'accueil </button>
</div>"
`;
